pkg/gardenlet/controller/backupbucket/backup_bucket_reconciler.go

```diff
@@ -98,7 +98,6 @@ func (r *reconciler) reconcileBackupBucket(
 		if err := controllerutils.AddFinalizers(ctx, r.gardenClient, backupBucket, gardencorev1beta1.GardenerName); err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to add finalizer: %w", err)
 		}
-		return reconcile.Result{}, nil
```

Comment:
It's a bit unfortunate that we now have to remove this after it only got added intentionally with https://github.com/gardener/gardener/pull/5683. OTOH, https://github.com/gardener/gardener/issues/4251 speaks about

> stop returning nil after adding finalizers in order to prevent duplicated reconciliations (basically implement option b) from https://github.com/gardener/gardener/issues/4730 where it is not done yet)

Option b) was 

> changing the update event handlers to enqueue objects only when the generation changed (instead of comparing metadata.generation and status.observedGeneration), this is also what controller-runtime supports out of the box with predicate.GenerationChangedPredicate

We should do this consistently everywhere then. I will add a dedicated task to https://github.com/gardener/gardener/issues/4251 so that we can track it.